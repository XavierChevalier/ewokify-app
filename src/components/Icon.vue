<script setup lang="ts">
  import { PropType } from 'vue'
  import { MdiExtra } from '@/assets/images/icons/MdiExtra'
  import { isPropertyValid } from '@/tools/component-properties/PropertyValidator'

  defineProps({
    path: {
      type: [String, Object] as PropType<MdiExtra>,
      required: true,
      // eslint-disable-next-line vue/valid-define-props
      validator: isPropertyValid(MdiExtra),
    },
    size: {
      type: Number,
      default: () => 24,
    },
  })
</script>

<template>
  <svg :width="size" :height="size" :viewBox="path.viewBox || '0 0 24 24'">
    <path
      :transform="path.transform"
      :d="path.path || path"
      style="fill: currentColor"
    />
  </svg>
</template>
