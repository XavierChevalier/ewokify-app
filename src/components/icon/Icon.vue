<script setup lang="ts">
  import { PropType } from 'vue'
  import { MdiExtra } from '@/assets/images/icons/MdiExtra'

  defineProps({
    path: {
      type: [String, Object] as PropType<MdiExtra>,
      required: true,
      // eslint-disable-next-line vue/valid-define-props
      validator: (value: MdiExtra) => MdiExtra.safeParse(value).success,
    },
    size: {
      type: Number,
      default: () => 24,
    },
  })
</script>

<template>
  <svg :width="size" :height="size" :viewBox="path.viewBox || '0 0 24 24'">
    <path
      :transform="path.transform"
      :d="path.path || path"
      style="fill: currentColor"
    />
  </svg>
</template>
