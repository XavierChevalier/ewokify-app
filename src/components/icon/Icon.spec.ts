import { mount } from '@vue/test-utils'
import Icon from './Icon.vue'
import { mdiAlert } from '@mdi/js'

describe('Icon', () => {
  it('should render the mdiAlert icon', () => {
    const wrapper = mount(Icon, {
      props: {
        path: mdiAlert,
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
