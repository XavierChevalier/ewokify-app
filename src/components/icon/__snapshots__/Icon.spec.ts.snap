// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render the mdiAlert icon 1`] = `
<svg
  height="24"
  viewBox="0 0 24 24"
  width="24"
>
  <path
    d="M13 14H11V9H13M13 18H11V16H13M1 21H23L12 2L1 21Z"
  />
</svg>
`;
